### ISSUE TITLE
tfjs-node support for loadSavedModel missing memory usage

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#4005]
Models loaded in `tfjs_node` using `loadSavedModel()` do not register their memory usage as (output of `tf.memory()`)  
Same model loaded using `loadGraphModel()` shows correct values

```js
const fs = require('fs');
const path = require('path');
const tf = require('@tensorflow/tfjs-node');
const { model } = require('@tensorflow/tfjs-node');
const { testEpsilon } = require('@tensorflow/tfjs-node/node_modules/@tensorflow/tfjs-core/dist/test_util');

async function main() {
  await tf.setBackend('tensorflow');
  await tf.enableProdMode();
  await tf.ready();
  const saved = await tf.node.loadSavedModel('ssd-mobilenet-v2');
  console.log('Memory:Saved', tf.memory());
  const graph = await tf.loadGraphModel(`file://ssd-mobilenet-v2-graph/model.json`);
  console.log('Memory:Graph', tf.memory());
}

main();
```

```text
Memory:Saved { unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
Memory:Graph { unreliable: true, numTensors: 202, numDataBuffers: 202, numBytes: 67334496 }
```text

This is the case for any model.  
For a test I've used `saved_model` and `graph_model` from <https://tfhub.dev/tensorflow/tfjs-model/ssd_mobilenet_v2/1/default/1>

*Environment: Ubuntu 20.04, NodeJS 14.11.0, TFJS built from master dated 09/30/2020*

**Comment by tafsiri** [comment]
I don't think you would necessarily expect to see tensors created at load time for the direct SavedModel execution provided by tf.node.loadSavedModel(). Direct SavedModel execution works a bit different from how we load tfjs graph models so no tensor memory may have been allocated at that point (and they also don't need to go through the JS layer to run the model).

After inference I would expect to see counts for output tensors. 

cc @pyu10055 in case my understanding is incorrect and this is not behaving as expected.

**Comment by vladmandic** [comment]
@tafsiri 

I just tried to get `tf.memory()` after call to `loadSavedModel()` and once more after call to `predict()`.
Yes, output is different, but it's equally incorrect - model has 200+ tensors, not 5. And `numbytes` value just makes no sense.

```text
2020-10-01 10:50:12 STATE:  TensorFlow/JS Memory after load: { unreliable: true, numTensors: 1, numDataBuffers: 1, numBytes: 234864000 }
2020-10-01 10:50:13 STATE:  TensorFlow/JS Memory after predict: { unreliable: true, numTensors: 5, numDataBuffers: 5, numBytes: 234866404 }
```

**Comment by tafsiri** [comment]
Yes in this case the tensors used during execution are not trackable at the JS layer (we pretty much can skip the js layer entirely when executing the model), This is why the 'unreliable' flag is set to true. Also I would expect intermediate tensors to be cleaned up by the end of the inference. 

Overall what are you trying to do? Maybe there is another path we can point you towards (it might be better in this case to use OS utilities to track memory). Or if you were just reporting the discrepancy so that we can take a look that is fine too!

**Comment by vladmandic** [comment]
@tafsiri My use cases are:
- Track memory usage as I'm loading multiple models as running out of memory is a real possibility.  
  Yes, I can use OS utilities for that, but that would capture everything inside the app which - better than nothing, but not ideal.
- Track any possible memory leaks

Big thing for me is that I have a large code-base already using `loadGraphModel()` and I'm forced to use `loadSavedModel()` for some models due to TF converter bugs (I've filled them separately) and I'd like to have code as uniform as possible.

**Comment by vladmandic** [comment]
Yes, definitely, although it's not a high priority item.

**Comment by avmohansai** [comment]
Hi @vladmandic ,
We are closing this issue due to inactivity. If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it.
Thank you for your understanding.
